web/src/components/Settings/SystemSection.tsx

```diff
@@ -104,18 +94,22 @@ const SystemSection = () => {
   };
 
   const handleOpenAIConfigKeyChanged = (value: string) => {
-    setOpenAIConfig({
-      ...openAIConfig,
-      key: value,
+    setState({
+      ...state,
+      openAIConfig: {
+        ...state.openAIConfig,
+        key: value,
+      },
     });
   };
 
   const handleSaveOpenAIConfig = async () => {
     try {
       await api.upsertSystemSetting({
         name: "openai-config",
-        value: JSON.stringify(openAIConfig),
+        value: JSON.stringify(state.openAIConfig),
       });
+      globalStore.setSystemStatus({ openAIConfig: state.openAIConfig });
```

Comment:
Added after request to backend to avoid mismatch with frontend in case of an error. It's done the other way round for `disablePublicMemos`, does anyone know is there a reason for that?

Reply:
I think both works. The frontend will "refresh" state before or after backend api request finished.